js/https.js

```diff
@@ -0,0 +1,171 @@
+            if (subdomain && subdomain !== 'www') {
+                const wildcard = host.replace(subdomain, '*')
+                loop.push(wildcard)               
+            }
+
+            // Check db
+            let isResolved = false
+            loop.forEach((r, i) => {
+                if (isResolved) return
+                this.db
+                    .get(this.dbObjectStore, r)
+                    .then(
+                        (record) => {
+                            if (record && record.simpleUpgrade) {
+                                const upgrade = reqUrl.replace(/^(http|https):\/\//, 'https://')
+                                isResolved = true
+                                return resolve(upgrade)
+                            }
+                            if (i === (loop.length - 1)) return resolve(reqUrl)
+                        },
+                        () => {
+                            console.warn('HTTPS: pipeRequestUrl() encountered a db error')
+                            if (i === (loop.length -1)) return resolve(reqUrl)
+                        }
+                    )
+
+            })
+
+        })
+    }
+
+    /* For debugging/development/test purposes only */
+    getHostRecord (host) {
+        return new Promise ((resolve, reject) => {
+            if (!this.isReady) {
+                console.warn('HTTPS: this.db is not ready')
+                return reject()
+            } 
+          
+            this.db.get(this.dbObjectStore, host).then(
+                (record) => {
+                    if (record) return resolve(record)
+                    return resolve()
+                },
+                () => {
+                    console.warn('HTTPS: getHostRecord() encountered a db error.')
+                    return reject()
+                }
+            )
+
+        })
+    }
+
+    /* For debugging/development/test purposes only */
+    testGetHostRecord () {
+        // These hosts should always have records that were xhr'd 
+        // into the client-side db from server
+        const testHosts = [
+            '1337x.to',
+            'submit.pandora.com',
+            '*.api.roblox.com',
+            'thump.vice.com',
+            'yts.ag'
+        ]
+
+        // Test that host records exist after db install from server
+        testHosts.forEach((host) => {
+            this.getHostRecord(host).then(
+                (record) => {
+                    if (record) {
+                        console.log('HTTPS: retrieved record for host: ' + host)
+                        console.log(record)
+                        return
+                    }
+                    console.warn('HTTPS: could not find record for host: ' + host)
+                },
+                () => console.error('HTTPS: testDBKeys() encountered a db error for host: ' + host)
+            )
+        })        
+    }
+
+    /* For debugging/development/test purposes only */
+    testPipeRequestUrl () {
```

Comment:
Do you want to add this as a unit test now that they work again?

Reply:
Yes. Will do.

Reply:
Hm, scoping is weird in /tests folder. Global dependencies blow up and I keep getting `defaultSettings` is not defined -- and unfortunately https needs `defaultSettings` to be defined (unlike some of the other tests, which are more standalone). I remember seeing this before the latest PR for /tests was merged but i didn't say anything but now am blocked on writing tests bc of it. Maybe I am misunderstanding something, check out screenshot. There's no tests for `Site` either, I wonder if its due to same reason? Would like to merge this and revisit in "Phase 3" of HTTPS Everywhere if there's no quick solution. IMO scoping throughout background of extension is kind of hinky due to legacy `require.scopes` being odd (I added HTTPS and IndexedDBClient to global instead of require.scopes).
![screen shot 2017-09-04 at 10 49 52 pm](https://user-images.githubusercontent.com/1278969/30047101-48c56822-91c4-11e7-9e24-bd5e42934abc.png)

Reply:
I pushed an update to add defaultSettings. When you have your test ready you need to include it in `test/html/unitTests.html`.

The site tests need to be completely redone since we switched from the Site container to sites being a property of each tab.

Reply:
Yeah I had my test in `test/html/unitTests.html`, that's why i could see that `defaultSettings` was hosed (it was never even reaching my `https` test).

Reply:
Will pick it up and see if your update worked for what i need...